src/test/run-pass/typeck-closure-to-unsafe-fn-ptr.rs

```diff
@@ -0,0 +1,7 @@
+unsafe fn call_unsafe(func: unsafe fn() -> ()) -> () {
+    func()
+}
+
+pub fn main() {
+    unsafe { call_unsafe(|| {}); }
```

Comment:
Can you extend the tests (in another file, `// compile-fail`) checking that:
```rust
let _: unsafe fn() = || /* do random stuff that requires `unsafe { ... }` */;
```
does not compile?